ldes-fragmentisers/ldes-fragmentisers-substring/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/SubstringFragmentationStrategy.java

```diff
@@ -21,49 +22,56 @@ public class SubstringFragmentationStrategy extends FragmentationStrategyDecorat
 	private final ObservationRegistry observationRegistry;
 	private final SubstringFragmentFinder substringFragmentFinder;
 	private final SubstringFragmentCreator substringFragmentCreator;
-	private final LocalMemberSupplier localMemberSupplier;
+	private final SubstringConfig substringConfig;
 
 	public SubstringFragmentationStrategy(FragmentationStrategy fragmentationStrategy,
 			ObservationRegistry observationRegistry,
 			SubstringFragmentFinder substringFragmentFinder,
 			SubstringFragmentCreator substringFragmentCreator,
 			TreeRelationsRepository treeRelationsRepository,
-			LocalMemberSupplier localMemberSupplier) {
+			SubstringConfig substringConfig) {
 		super(fragmentationStrategy, treeRelationsRepository);
 		this.observationRegistry = observationRegistry;
 		this.substringFragmentFinder = substringFragmentFinder;
 		this.substringFragmentCreator = substringFragmentCreator;
-		this.localMemberSupplier = localMemberSupplier;
+		this.substringConfig = substringConfig;
 	}
 
 	@Override
 	public void addMemberToFragment(LdesFragment parentFragment, Member member, Observation parentObservation) {
-		final Observation substringFragmentationObservation = startFragmentationObservation(parentObservation);
 
-		final LocalMember localMember = localMemberSupplier.toLocalMember(member);
+		final Observation substringFragmentationObservation = startFragmentationObservation(parentObservation);
+		FragmentationString fragmentationString = new FragmentationString(getFragmentationString(member));
 		final LdesFragment rootFragment = prepareRootFragment(parentFragment);
-		addMemberToFragments(parentFragment, localMember, substringFragmentationObservation, rootFragment);
+		Set<LdesFragment> ldesFragments = getSubstringFragments(parentFragment, fragmentationString, rootFragment);
+		ldesFragments.parallelStream()
+				.forEach(substringFragment -> super.addMemberToFragment(substringFragment, member,
+						substringFragmentationObservation));
 		substringFragmentationObservation.stop();
 	}
 
+	private String getFragmentationString(Member member) {
```

Comment:
I don't feel that moving this logic here is a big improvement as this class now has to know a lot more about the internals of the classes and its constructing and initialization logic. It moves more to the anemic services anti pattern ;)